spec/models/manageiq/providers/azure/cloud_manager/event_catcher_spec.rb

```diff
@@ -3,20 +3,19 @@
     let(:unsupported_reason) { "Timeline events not supported for this region" }
 
     before do
-      @ems = FactoryBot.create(:ems_azure)
-      allow(@ems).to receive(:authentication_status_ok?).and_return(true)
-      allow(described_class).to receive(:all_ems_in_zone).and_return([@ems])
+      server = EvmSpecHelper.local_miq_server
+      @ems = FactoryGirl.create(:ems_azure, :with_authentication, :zone => server.zone)
     end
 
-    it "returns a valid ems for zone if timeline events are supported" do
-      allow(@ems).to receive(:supports_timeline?).and_return(true)
+    it "returns a valid ems for zone when timeline events are supported" do
+      expect_any_instance_of(@ems.class).to receive(:supports_timeline?).and_return(true)
```

Comment:
a. I don't know if these tests are even valid.
I see them in other places, but since this method was not defined here, seems moot.
b. Is it possible for an azure ems to not support timelines?

c. rubocop does not like this. Not sure that the complain should stop us from merging but...

The alternative is to not stub it at all. Does anyone know how to setup an ems so it either does or does not `supports_timeline?` ? (again, is this even valid?)

Reply:
Regarding `b)`, it's possible if certain services are disabled on the subscription. I don't see that ever happening in practice though. Stack is another matter, but that's a separate provider now.